dali/test/python/test_operator_sequence_rearrange.py

```diff
@@ -82,22 +82,38 @@ def test_sequence_rearrange():
                 for layout in ["FHW"[:len(shape)], ""]:
                     yield check_sequence_rearrange, 5, shape, new_order, per_sample, dev, layout
 
-@raises(RuntimeError)
 def check_fail_sequence_rearrange(batch_size, shape, reorders, persample_reorder=True, op_type="cpu", layout=""):
     check_sequence_rearrange(batch_size, shape, reorders, persample_reorder, op_type, layout)
 
 def test_fail_sequence_rearrange():
+    shape = [5, 1]
+    orders = [
+        ([6, 7], False),
+        ([-1], False),
+        ([], False),
+        ([np.int32([0]), np.int32([])], True),
+        ([np.int32([6, 7]), np.int32([0])], True),
+        ([np.int32([-1]), np.int32([0])], True),
+        ([np.int32([[1], [2]]), np.int32([[1], [2]])], True)
+    ]
+    error_msgs = [
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'Empty result sequences are not allowed',
+        'Empty `new_order` for sample * is not allowed',
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'Input with dimension * cannot be converted to dimension *'
+    ]
+
     for dev in ["cpu", "gpu"]:
-        for shape in [[5, 1]]:
-            for new_order, per_sample in [([6, 7], False), ([-1], False), ([], False),
-                    ([np.int32([0]), np.int32([])], True), ([np.int32([6, 7]), np.int32([0])], True),
-                    ([np.int32([-1]), np.int32([0])], True), ([np.int32([[1], [2]]), np.int32([[1], [2]])], True)]:
-                yield check_fail_sequence_rearrange, 2, shape, new_order, per_sample, dev
+        for args, error_msg in zip(orders, error_msgs):
+            yield raises(RuntimeError, glob=error_msg)(check_fail_sequence_rearrange), 2, shape, *args, dev
 
 def test_wrong_layouts_sequence_rearrange():
     shape = [5, 1]
     new_order = [0, 2, 1, 3, 4]
     per_sample = False
     for dev in ["cpu", "gpu"]:
         for layout in ["HF", "HW"]:
-            yield check_fail_sequence_rearrange, 5, shape, new_order, per_sample, dev, layout
+            yield raises(RuntimeError, glob="Expected sequence as the input, where outermost dimension represents frames dimension `F`")(check_fail_sequence_rearrange), 5, shape, new_order, per_sample, dev, layout
```

Comment:
That's also a bit cryptic to an outsider (maybe mentioning layout would help?).

Reply:
Done